### ISSUE TITLE
Odd lack of basic documentation?

### ISSUE THREAD

**Issue by alxpettit** [simh/simh#630]
I see lots of in-depth installation information, which is great, but I honestly can't figure out how I'm supposed to use SIMH. Please note that I'm a complete noob in this area, and circa 1970s machine architecture was well before my time. (But I still want to learn about it, hence why I'm here!)

My intention is to simulate a VAX machine for running [2 BSD](https://archive.org/details/2bsd.tar). I have skimmed the readme files and the `install.ms` _(incidentally, can anyone tell me what this `ms` format is? I don't recognize the extension or the encoding, but it seems to be some sort of rich text formatting)_ -- and I was not really able to apply those instructions to a modern architecture or simulator. 

I'm not even sure what kind of BIOS these machines had, or otherwise which file they load into memory.

And I oddly enough can't seem to find the kernel file in the 2 BSD package, but my sources seem to imply it's a complete operating system rather than a suite of tools (correct me if I'm wrong, there).

So, in short, I have no idea what I'm doing, and would appreciate some basic beginner-friendly instructions for someone adept with modern systems, but completely unfamiliar with historical architectures. Thanks! ^.^

P.S.: it appears that the [SIMH FAQs](http://simh.trailing-edge.com/simh_faq.txt) link is broken. I thought someone would want to know about that!

**Comment by markpizz** [comment]
Hi Alexandria,

I'm glad you were able to find your way to the GitHub simh resource.  The latest active development for simh goes on here and this is the repository of the most up to date documentation.  The documentation for each simh simulator can be found (in Microsoft Doc format) in the doc directory of the repository (https://github.com/simh/simh/tree/master/doc).  The most up to date FAQ is located there along with how to use the simh simulators in simh_doc.doc,

The link you've pointed to for 2 BSD, specifically refers to running on a PDP11, which fortunately is one of the simulators available here.  There is some mention of the later versions of 2 BSD getting pieces back from the later versions of VAX BSD versions.  Running either BSD version could be choices,   The recipes that get these things to work come from a combination of the original installation and setup documents from when these operating systems shipped.  Since these things come from so long ago, and since many folks have preceded you, some of those folks worked out the details and wrote down what they did.  If you're working from one of those.  Most of these recipes are found proximate to the operating systems that are being installed.  This is due to the community of folks interested in those operating systems were the ones who did the detailed leg work.  Most of the interest by folks like you is really about getting these operating systems running so you can explore the details of the operating system history, and not have to fuss with what may be otherwise obtuse or obscure details about using simh.  If, as you move forward with your efforts, you encounter problematic behavior from a simh simulator, please come back with specific questions or problem details.

Meanwhile, looking back at your original question and looking closely at the files your 2 BSD link points at, it seems that what you've got there is a combination source and binary distribution of the OS.  It seems that it's intention was not to perform an install to new hardware, but as a means of upgrading a previous version of the OS.  The files with .ms extensions seem to merely be files with some old form of a markup language for text documents.  They are completely text files which are easily read.  If you open the READ_ME file, it explains how to run nroff on the install.ms file to generate the formatted result.

This is about as much non simh digging as I'm willing to do.  Hopefully you can find a full recipe that describes the beginning to ending process to get this working on under a simh simulator.

Good Luck,

- Mark Pizzolato

**Comment by larsbrinkhoff** [comment]
Here are some guides:

- Installing 2.9BSD onto a PDP-11: http://gunkies.org/wiki/Installing_2.9_BSD_on_SIMH
- Installing 3BSD onto a VAX: http://gunkies.org/wiki/Installing_3_BSD_on_SIMH

**Comment by jguillaumes** [comment]
Dockerized, ready to run simh-pdp11 + 2BSD:

jguillaumes/simh-pdpbsd in hub.docker.com


Jordi Guillaumes Pons


El 8 oct 2018, a les 11:46, Lars Brinkhoff <notifications@github.com> va escriure:

**Comment by alxpettit** [comment]
Thanks! Maybe you should add these examples to the README.md, so that beginners can find them?

**Comment by Alhadis** [comment]
> *incidentally, can anyone tell me what this `ms` format is? I don't recognize the extension or the encoding, but it seems to be some sort of rich text formatting*

`ms` stands for "manuscript", one of the standard macro packages distributed with Troff. Troff (short for "**T**ypesetter **R**un**off**", pronounced _“tee-roff”_) is a typesetting program best known as the formatter of `man` pages (to a point where some folks mistake `man` as its own markup language). Think of it as an even older, hairier version of [TeX](https://en.wikipedia.org/wiki/TeX).

You can view `*.ms` or any other Roff documents like this:

~~~shell
# Modern (GNU) troff (a.k.a., groff)
groff -Tutf8 -ms /path/to/file.ms | less            # Short form
troff -Tutf8 -ms /path/to/file.ms | grotty | less   # Long form

# Older troff(1) implementations
nroff -ms /path/to/file.ms | less
~~~

The full list of standard macro packages you'll find in the wild are:

* `man` — **Man**ual pages
* `ms` — **M**anu**s**cripts
* `mm` — **M**e**m**orandums
* `me` — **E**ric Allman's macros †
* `mdoc` — BSD-flavoured semantic alternative to the original `man` macros
* `mom` — "**M**y **O**wn **M**acros", a general-purpose typesetting package for GNU Troff

† Notice how they all start with the letter M? That's because Troff's switch for selecting a macro package is `-m`, and the argument it took was the name of a `tmac` (Troff macro) file. Early authors abused the fact that whitespace was optional and gave `tmac` files names like `an.tmac` or `s.tmac` so they could be selected using `-man` or `-ms`, respectively (analoguous to `-m an` or `-m s`, which isn't as readable). Unfortunately this trick doesn't work as well when the package's name begins with another letter, yielding odd-looking names like `mdoc` or `me`. 

^Z

**Comment by larsbrinkhoff** [comment]
^Z as in suspend, or end of file?  ITS users want to know.

**Comment by Alhadis** [comment]
End-of-file. 😂 I was hoping somebody would pick up on that.
